src/index.ts

```diff
@@ -11,6 +11,12 @@ export interface PromiseOptions {
   readonly timeoutMs?: number;
 }
 
+export class GoWrappedError extends Error {
+  constructor(public cause: unknown) {
+    super('' + cause);
+  }
+}
+
```

Comment:
Ok, this confused me a bit. First of all, I didn't know that you can define a property in the constructor in TS :smile: But calling the property `cause` is a bit confusing because you can actually [pass the `cause` option to the Error constructor](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Error#rethrowing_an_error_with_a_cause) (which is not happening here) and that confused me :smile: I'll leave up to you if you want to change the name.

Reply:
I can rename it to `reason`.